server/src/main/java/org/apache/druid/server/coordinator/CoordinatorDynamicConfig.java

```diff
@@ -192,6 +212,35 @@ public int getMaxSegmentsInNodeLoadingQueue()
     return maxSegmentsInNodeLoadingQueue;
   }
 
+  /**
+   * Historical nodes list in maintenance mode. Coordinator doesn't assign new segments on those nodes and moves
+   * segments from those nodes according to a specified priority.
+   *
+   * @return list of host:port entries
+   */
+  @JsonProperty
+  public Set<String> getHistoricalNodesInMaintenance()
+  {
+    return historicalNodesInMaintenance;
+  }
+
+  /**
+   * Priority of segments from servers in maintenance. Coordinator takes ceil(maxSegmentsToMove * (priority / 10))
+   * from servers in maitenance during balancing phase, i.e.:
+   * 0 - no segments from servers in maintenance will be processed during balancing
+   * 5 - 50% segments from servers in maintenance
+   * 10 - 100% segments from servers in maintenance
+   * By leveraging the priority an operator can prevent general nodes from overload or decrease maitenance time
+   * instead.
+   *
+   * @return number in range [0, 10]
```

Comment:
why not just 1-100 if it represents a %?

Reply:
The initial term was a priority, which is usually not a big number (eg, Java thread priority). 
Percentage term emerged during writing that description. Moreover, I don't expect someone would like to define 65%, 75% or even 67%, 23% and so on, usually 10, 20, ..., 80 is good enough, and then adding 0 every time seems redudant. Meanwhile a priority term allows to be more concise.